src/Platform/Microsoft.Testing.Platform/UI/ConsoleLoggerOutputDevice.cs

```diff
@@ -0,0 +1,511 @@
+                                break;
+                            case ConsoleColor.Yellow:
+                                _consoleLogger.WriteWarningMessage(_assemblyName, _targetFramework, _architecture, formattedTextOutputDeviceData.Text);
+                                break;
+                            default:
+                                _consoleLogger.WriteMessage(formattedTextOutputDeviceData.Text);
+                                break;
+                        }
+                    }
+                    else
+                    {
+                        _consoleLogger.WriteMessage(formattedTextOutputDeviceData.Text);
+                    }
+
+                    break;
+
+                case TextOutputDeviceData textOutputDeviceData:
+                    {
+                        await LogDebugAsync(textOutputDeviceData.Text);
+                        _consoleLogger.WriteMessage(textOutputDeviceData.Text);
+                        break;
+                    }
+
+                case ExceptionOutputDeviceData exceptionOutputDeviceData:
+                    {
+                        await LogDebugAsync(exceptionOutputDeviceData.Exception.ToString());
+                        _consoleLogger.WriteErrorMessage(_assemblyName, _targetFramework, _architecture, exceptionOutputDeviceData.Exception);
+
+                        break;
+                    }
+            }
+        }
+    }
+
+    public async Task ConsumeAsync(IDataProducer dataProducer, IData value, CancellationToken cancellationToken)
+    {
+        await Task.CompletedTask;
+        if (cancellationToken.IsCancellationRequested)
+        {
+            return;
+        }
+
+        switch (value)
+        {
+            case TestNodeUpdateMessage testNodeStateChanged:
+
+                TimeSpan duration = testNodeStateChanged.TestNode.Properties.SingleOrDefault<TimingProperty>()?.GlobalTiming.Duration ?? TimeSpan.Zero;
+                switch (testNodeStateChanged.TestNode.Properties.SingleOrDefault<TestNodeStateProperty>())
+                {
+                    case InProgressTestNodeStateProperty:
+                        // do nothing.
+                        break;
+
+                    case ErrorTestNodeStateProperty errorState:
+                        _consoleLogger.TestCompleted(
+                         assembly: _assemblyName,
+                         targetFramework: _targetFramework,
+                         architecture: _architecture,
+                         displayName: testNodeStateChanged.TestNode.DisplayName,
+                         outcome: LoggerOutcome.Error,
+                         duration: duration,
+                         errorMessage: errorState.Exception?.Message ?? errorState.Explanation,
+                         errorStackTrace: errorState.Exception?.StackTrace,
+                         expected: null,
+                         actual: null);
```

Comment:
Arguments are not indented correctly (that's the case for all the other entries of the switch).

Reply:
formated removed in places where obvious value was provided.